internal/c/parts/gui/tinyfiledialogs.h

```diff
@@ -0,0 +1,303 @@
+		/* aoResultRGB also contains the result */
+		/* aDefaultRGB is used only if aDefaultHexRGB is NULL */
+		/* aDefaultRGB and aoResultRGB can be the same array */
+		/* returns NULL on cancel */
+
+#endif /*_WIN32 */
+
+#ifdef	__cplusplus
+} /*extern "C"*/
+#endif
+
+#endif /* TINYFILEDIALOGS_H */
+
+/*
+ ________________________________________________________________________________
+|  ____________________________________________________________________________  |
+| |                                                                            | |
+| | on windows:                                                                | |
+| |  - for UTF-16, use the wchar_t functions at the bottom of the header file  | |
+| |  - _wfopen() requires wchar_t                                              | |
+| |                                                                            | |
+| |  - in tinyfiledialogs, char is UTF-8 by default (since v3.6)               | |
+| |  - but fopen() expects MBCS (not UTF-8)                                    | |
+| |  - if you want char to be MBCS: set tinyfd_winUtf8 to 0                    | |
+| |                                                                            | |
+| |  - alternatively, tinyfiledialogs provides                                 | |
+| |                        functions to convert between UTF-8, UTF-16 and MBCS | |
+| |____________________________________________________________________________| |
+|________________________________________________________________________________|
+
+- This is not for ios nor android (it works in termux though).
+- The files can be renamed with extension ".cpp" as the code is 100% compatible C C++
+  (just comment out << extern "C" >> in the header file)
+- Windows is fully supported from XP to 10 (maybe even older versions)
+- C# & LUA via dll, see files in the folder EXTRAS
+- OSX supported from 10.4 to latest (maybe even older versions)
+- Do not use " and ' as the dialogs will be displayed with a warning
+  instead of the title, message, etc...
+- There's one file filter only, it may contain several patterns.
+- If no filter description is provided,
+  the list of patterns will become the description.
+- On windows link against Comdlg32.lib and Ole32.lib
+  (on windows the no linking claim is a lie)
+- On unix: it tries command line calls, so no such need (NO LINKING).
+- On unix you need one of the following:
+  applescript, kdialog, zenity, matedialog, shellementary, qarma, yad,
+  python (2 or 3)/tkinter/python-dbus (optional), Xdialog
+  or curses dialogs (opens terminal if running without console).
+- One of those is already included on most (if not all) desktops.
+- In the absence of those it will use gdialog, gxmessage or whiptail
+  with a textinputbox. If nothing is found, it switches to basic console input,
+  it opens a console if needed (requires xterm + bash).
+- for curses dialogs you must set tinyfd_allowCursesDialogs=1
```

Comment:
I mentioned it in the discord but I think we should really consider enabling this for `$CONSOLE` programs. The comments in the code suggest that we can either enable it as a fallback in the event the graphics-based dialogs don't work, or always force the console-based dialog, and also I think we can more or less change it at any time.

I'm not entirely sure yet, but perhaps we should determine whether the console-based dialog is allowed/forced based on whether `_CONSOLE ON` is currently enabled. That's always on for `$CONSOLE:ONLY`, and would optionally be on for programs like QB64-PE which only use console output in some circumstances. I think that gives the most natural result, as it means using something like `qb64pe` gives regular graphical dialogs like expected, but using `qb64pe -x` will always give console-based dialogs if they come up for some reason (Ex. error).

Regardless of the exact semantics, allowing console dialogs would be nice because it lets the user see error pop-up's or these dialogs even if their program is console-based and there is no graphics server running (which is common on Linux).

Reply:
Ok. We can set `tinyfd_allowCursesDialogs=1` when `$CONSOLE:ONLY` is specified on Linux. Or should we do it for all platforms? There is a `tinyfd_forceConsole` also. I'll also try a couple of things and share my findings.

Reply:
I think it depends how well the console dialogs work, I suspect the Linux ones are better than the Windows ones. Console-only Windows (no graphics at all) is also significantly less common than for Linux (in my experience anyway), so it's probably less of a concern for Windows.

We'll probably want to do a bit with separate `cpp`s to initialize that setting without using `#ifdef`s in `libqb.cpp`, after we determine how we want it to function I could help with that. Also we could potentially leave this for another card if you want.

Reply:
I am going to leave this for you. 😁 I cannot call myself any level of expert on Linux. I think we should check the behavior first though. I have no (or don't know a) way of testing this currently. I want to know what tfd does if we run `$CONSOLE:ONLY` stuff on Linux without X.

Reply:
Done, I made #219 for it :)